launcher-common/src/main/java/org/apache/brooklyn/launcher/command/support/CloudExplorerSupport.java

```diff
@@ -0,0 +1,437 @@
+    public boolean autoconfirm = false;
+
+
+    @VisibleForTesting
+    protected PrintStream stdout = System.out;
+
+    @VisibleForTesting
+    protected PrintStream stderr = System.err;
+
+    @VisibleForTesting
+    protected InputStream stdin = System.in;
+
+    public CloudExplorerSupport(ManagementContext managementContext, boolean allLocations, String location, boolean autoconfirm) {
+        this.managementContext = managementContext;
+        this.allLocations = allLocations;
+        this.location = location;
+        this.autoconfirm = autoconfirm;
+    }
+
+    protected abstract void doCall(JcloudsLocation loc, String indent) throws Exception;
+
+    @Override
+    public Void call() throws Exception {
+        List<JcloudsLocation> locs = Lists.newArrayList();
+        if (location != null && allLocations) {
+            throw new FatalConfigurationRuntimeException("Must not specify --location and --all-locations");
+        } else if (location != null) {
+            JcloudsLocation loc = (JcloudsLocation) managementContext.getLocationRegistry().getLocationManaged(location);
+            locs.add(loc);
+        } else if (allLocations) {
+            // Find all named locations that point at different target clouds
+            Map<String, LocationDefinition> definedLocations = managementContext.getLocationRegistry().getDefinedLocations();
+            for (LocationDefinition locationDef : definedLocations.values()) {
+
+                Location loc = managementContext.getLocationManager().createLocation(
+                    managementContext.getLocationRegistry().getLocationSpec(locationDef).get() );
+
+                if (loc instanceof JcloudsLocation) {
+                    boolean found = false;
+                    for (JcloudsLocation existing : locs) {
+                        if (equalTargets(existing, (JcloudsLocation) loc)) {
+                            found = true;
+                            break;
+                        }
+                    }
+                    if (!found) {
+                        locs.add((JcloudsLocation) loc);
+                    }
+                }
+            }
+        } else {
+            throw new FatalConfigurationRuntimeException("Must specify one of --location or --all-locations");
+        }
+
+        for (JcloudsLocation loc : locs) {
+            stdout.println("Location {");
+            stdout.println("\tprovider: "+loc.getProvider());
+            stdout.println("\tdisplayName: "+loc.getDisplayName());
+            stdout.println("\tidentity: "+loc.getIdentity());
+            if (loc.getEndpoint() != null) stdout.println("\tendpoint: "+loc.getEndpoint());
+            if (loc.getRegion() != null) stdout.println("\tregion: "+loc.getRegion());
+
+            try {
+                doCall(loc, "\t");
+            } finally {
+                stdout.println("}");
+            }
+        }
```

Comment:
This could be simplified by using `org.apache.karaf.shell.support.table.ShellTable`

Unless original formatting is required to remain the same